src/DataShare/DataShare/Trigger/NewAzDataShareTrigger.cs

```diff
@@ -0,0 +1,139 @@
+{
+    using System;
+    using Microsoft.Azure.Commands.DataShare.Common;
+    using Microsoft.Azure.Commands.DataShare.Helpers;
+    using Microsoft.Azure.Management.DataShare;
+    using Microsoft.Azure.Management.DataShare.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+    using System.Management.Automation;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Properties;
+
+    /// <summary>
+    /// Defines the New-DataShareTrigger cmdlet.
+    /// </summary>
+    [Cmdlet(
+         "New",
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareTrigger",
+         SupportsShouldProcess = true,
+         DefaultParameterSetName = ParameterSetNames.FieldsParameterSet), OutputType(typeof(PSDataShareInvitation))]
+    public class NewAzDataShareTrigger : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name of the azure data share account",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Azure data share account name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// Name of the azure data share subscription.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Azure data share subscription name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.ShareSubscription, "ResourceGroupName", "AccountName")]
+        public string ShareSubscriptionName { get; set; }
+
+        /// <summary>
+        /// Name of the azure data share trigger.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Azure data share trigger name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+
+        /// <summary>
+        /// Interval at which to synchronize the data share.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
No ValueFromPipelineByPropertyName for any parameter other than ResourceId.  Apply this rul universally.